File: sdkwork-ai-sdk-flutter/lib/src/api/threads.dart
import '../http/client.dart';
import '../models.dart';

class ThreadsApi {
  final HttpClient _client;
  
  ThreadsApi(this._client);

  /// Submit tool outputs
  Future<SubmitToolOutputsResponse?> submitToolOutputs(String thread_id, String run_id, SubmitToolOutputsRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}/submit_tool_outputs'), body: body, contentType: 'application/json');
    return response is SubmitToolOutputsResponse ? response : null;
  }

  /// Cancel run
  Future<RunResponse?> cancelRun(String thread_id, String run_id) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}/cancel'));
    return response is RunResponse ? response : null;
  }

  /// Get run
  Future<RunResponse?> getRun(String thread_id, String run_id) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}'));
    return response is RunResponse ? response : null;
  }

  /// Modify run
  Future<RunResponse?> updateRun(String thread_id, String run_id, UpdateRunRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}'), body: body, contentType: 'application/json');
    return response is RunResponse ? response : null;
  }

  /// List runs
  Future<RunListResponse?> listRuns(String thread_id, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/runs'), params: params);
    return response is RunListResponse ? response : null;
  }

  /// Create run
  Future<CreateRunResponse?> createRun(String thread_id, RunCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/runs'), body: body, contentType: 'application/json');
    return response is CreateRunResponse ? response : null;
  }

  /// Get thread message
  Future<ThreadMessageResponse?> getThreadMessage(String thread_id, String message_id) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/messages/${message_id}'));
    return response is ThreadMessageResponse ? response : null;
  }

  /// Modify thread message
  Future<ThreadMessageResponse?> updateThreadMessage(String thread_id, String message_id, ThreadMessageModifyRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/messages/${message_id}'), body: body, contentType: 'application/json');
    return response is ThreadMessageResponse ? response : null;
  }

  /// List thread messages
  Future<ThreadMessageListResponse?> listThreadMessages(String thread_id, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/messages'), params: params);
    return response is ThreadMessageListResponse ? response : null;
  }

  /// Create thread message
  Future<ThreadMessageResponse?> createThreadMessage(String thread_id, ThreadMessageCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}/messages'), body: body, contentType: 'application/json');
    return response is ThreadMessageResponse ? response : null;
  }

  /// Get thread
  Future<ThreadResponse?> getThread(String thread_id) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}'));
    return response is ThreadResponse ? response : null;
  }

  /// Update thread
  Future<ThreadResponse?> updateThread(String thread_id, ThreadCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/${thread_id}'), body: body, contentType: 'application/json');
    return response is ThreadResponse ? response : null;
  }

  /// Delete thread
  Future<ThreadDeleteResponse?> deleteThread(String thread_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/threads/${thread_id}'));
    return response is ThreadDeleteResponse ? response : null;
  }

  /// Create thread and run
  Future<CreateThreadAndRunResponse?> createThreadAndRun(ThreadRunCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads/runs'), body: body, contentType: 'application/json');
    return response is CreateThreadAndRunResponse ? response : null;
  }

  /// Create thread
  Future<ThreadResponse?> createThread(ThreadCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/threads'), body: body, contentType: 'application/json');
    return response is ThreadResponse ? response : null;
  }

  /// Get run step
  Future<RunStepResponse?> getRunStep(String thread_id, String run_id, String step_id) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}/steps/${step_id}'));
    return response is RunStepResponse ? response : null;
  }

  /// List run steps
  Future<RunStepListResponse?> listRunSteps(String thread_id, String run_id, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/threads/${thread_id}/runs/${run_id}/steps'), params: params);
    return response is RunStepListResponse ? response : null;
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/threads.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import CreateRunResponse, CreateThreadAndRunPostResponse, RunCreateRequest, RunListResponse, RunResponse, RunStepListResponse, RunStepResponse, SubmitToolOutputsRequest, SubmitToolOutputsResponse, ThreadCreateRequest, ThreadDeleteResponse, ThreadMessageCreateRequest, ThreadMessageListResponse, ThreadMessageModifyRequest, ThreadMessageResponse, ThreadResponse, ThreadRunCreateRequest, UpdateRunPostRequest

class ThreadsApi:
    """Threads API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def submit_tool_outputs(self, thread_id: str, run_id: str, body: SubmitToolOutputsRequest) -> SubmitToolOutputsResponse:
        """Submit tool outputs"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}/submit_tool_outputs", json=body)

    def cancel_run(self, thread_id: str, run_id: str) -> RunResponse:
        """Cancel run"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}/cancel")

    def get_run(self, thread_id: str, run_id: str) -> RunResponse:
        """Get run"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}")

    def update_run(self, thread_id: str, run_id: str, body: UpdateRunPostRequest) -> RunResponse:
        """Modify run"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}", json=body)

    def list_runs(self, thread_id: str, params: Optional[Dict[str, Any]] = None) -> RunListResponse:
        """List runs"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/runs", params=params)

    def create_run(self, thread_id: str, body: RunCreateRequest) -> CreateRunResponse:
        """Create run"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/runs", json=body)

    def get_thread_message(self, thread_id: str, message_id: str) -> ThreadMessageResponse:
        """Get thread message"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/messages/{message_id}")

    def update_thread_message(self, thread_id: str, message_id: str, body: ThreadMessageModifyRequest) -> ThreadMessageResponse:
        """Modify thread message"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/messages/{message_id}", json=body)

    def list_thread_messages(self, thread_id: str, params: Optional[Dict[str, Any]] = None) -> ThreadMessageListResponse:
        """List thread messages"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/messages", params=params)

    def create_thread_message(self, thread_id: str, body: ThreadMessageCreateRequest) -> ThreadMessageResponse:
        """Create thread message"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}/messages", json=body)

    def get_thread(self, thread_id: str) -> ThreadResponse:
        """Get thread"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}")

    def update_thread(self, thread_id: str, body: ThreadCreateRequest) -> ThreadResponse:
        """Update thread"""
        return self._client.post(f"/ai/v3/v1/threads/{thread_id}", json=body)

    def delete_thread(self, thread_id: str) -> ThreadDeleteResponse:
        """Delete thread"""
        return self._client.delete(f"/ai/v3/v1/threads/{thread_id}")

    def create_thread_and_run(self, body: ThreadRunCreateRequest) -> CreateThreadAndRunPostResponse:
        """Create thread and run"""
        return self._client.post(f"/ai/v3/v1/threads/runs", json=body)

    def create_thread(self, body: ThreadCreateRequest) -> ThreadResponse:
        """Create thread"""
        return self._client.post(f"/ai/v3/v1/threads", json=body)

    def get_run_step(self, thread_id: str, run_id: str, step_id: str) -> RunStepResponse:
        """Get run step"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}/steps/{step_id}")

    def list_run_steps(self, thread_id: str, run_id: str, params: Optional[Dict[str, Any]] = None) -> RunStepListResponse:
        """List run steps"""
        return self._client.get(f"/ai/v3/v1/threads/{thread_id}/runs/{run_id}/steps", params=params)
